
///////////////////////////////////////////////////////////////////////////////
//
//  Copyright (c) 2009, Arizona State University
//  All rights reserved.
//  BSD License: http://www.opensource.org/licenses/bsd-license.html
//  Author: Jeff Conner
//
///////////////////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////////////////
//
//  Input dialog for files.
//
///////////////////////////////////////////////////////////////////////////////

#include "BuildingDialog.h"

#include "Usul/Strings/Format.h"

///////////////////////////////////////////////////////////////////////////////
//
//  Constructor.
//
///////////////////////////////////////////////////////////////////////////////

BuildingDialog::BuildingDialog ( QWidget *parent ) : 
BaseClass ( parent )
{
  // Initialize code from Designer.
  this->setupUi ( this );
  

  //QStringList titles;

  //titles.push_back( "X" );
  //titles.push_back( "Y" );
  //titles.push_back( "Z" );
  //titles.push_back( "Materials" );

  //_inputList->setHorizontalHeaderLabels( titles );

  ////initialize the table view
  //this->_initialize();
}


///////////////////////////////////////////////////////////////////////////////
//
//  Destructor.
//
///////////////////////////////////////////////////////////////////////////////

BuildingDialog::~BuildingDialog()
{
  
}


///////////////////////////////////////////////////////////////////////////////
//
//  Get the Building Parameters
//
///////////////////////////////////////////////////////////////////////////////

BuildingDialog::Building BuildingDialog::building()
{
  Building b( _lengthField->text().toStdString(), _widthField->text().toStdString(), _heightField->text().toStdString(), "0", "0", "0" );

  return b;
}


///////////////////////////////////////////////////////////////////////////////
//
//  Yes/No there is a building present
//
///////////////////////////////////////////////////////////////////////////////

bool BuildingDialog::useBuilding()
{
  bool value ( _buildingExists->checkState() );
  return value;
}